tool/smarts/src/test/java/org/openscience/cdk/smirks/SmirksTest.java

```diff
@@ -114,6 +121,35 @@ void testDuplicateMapIdx_Reactant() {
         assertEquals(transform.message(), "Duplicate atom map [C:4] and [O:4]");
     }
 
+    @Test
+    void testNegativeMapIdx() {
+        final String smirks = "[*:1][N:2](=[O:3])=[O:-4]>>[*:1][N+:2](=[O:3])-[OH:-4]";
+        SmirksTransform transform = new SmirksTransform();
+        transform.setPrepare(false);
+        assertFalse(Smirks.parse(transform, smirks), transform.message());
+        assertEquals("Invalid atom expression", transform.message());
+    }
+
+    @Disabled("I would expect Smirks.parse(transform, smirks) to fail with a MapIdx value of zero...?")
```

Comment:
https://www.daylight.com/dayhtml/doc/theory/theory.smirks.html

search for ":0]"

it is interpreted as unmapped, rather than fail this should be a warning.

Reply:
Wow, that comes as a surprise, good to know 🧠 🤓

Reply:
Well it's actually a mistake in the doc, Roger thinks it's because Dave W told someone how to do it and someone else wrote the doc :-). Anyways I've now made this an error, I would prefer a warning but it is dubious at best, but probably wrong.